backend/canisters/group/impl/src/lib.rs

```diff
@@ -100,6 +104,45 @@ impl RuntimeState {
         }
     }
 
+    pub fn queue_access_gate_payments(&mut self, gate: PaymentGate) {
+        // The amount available is the gate amount less the approval fee and the transfer_from fee
+        let amount_available = gate.amount - 2 * gate.fee;
+        // Queue a payment to each owner less the fee
+        let owners: Vec<UserId> = self
+            .data
+            .chat
+            .members
+            .iter()
+            .filter(|m| matches!(m.role.value, GroupRoleInternal::Owner))
+            .map(|m| m.user_id)
+            .collect();
+
+        let owner_count = owners.len() as u128;
+        let owner_share = (amount_available * 4 / 5) / owner_count;
+
+        for owner in owners {
+            self.data.pending_payments_queue.push(PendingPayment {
+                amount: owner_share - gate.fee,
+                fee: gate.fee,
+                ledger_canister: gate.ledger_canister_id,
+                recipient: PaymentRecipient::Member(owner),
+                reason: PendingPaymentReason::AccessGate,
+            });
+        }
+
+        // Queue the remainder to the treasury less the fee
+        let treasury_share = amount_available - (owner_share * owner_count);
```

Comment:
`saturating_sub` just to be safe? There could be 200 owners and this would underflow.